library/spdm_crypt_lib/fips/libspdm_selftest_rsa_ssa.c

```diff
@@ -0,0 +1,208 @@
+        0x4e, 0xe4, 0xee, 0x30, 0x9e, 0xb2, 0x5e, 0xb9,
+        0x7c, 0x75, 0xe1, 0xba, 0x60, 0x49, 0x94, 0x9b,
+        0xd6, 0xa7, 0xe1, 0x9e, 0x24, 0xbc, 0x9e, 0x37,
+        0xcf, 0x07, 0xea, 0xa8, 0xaf, 0xcc, 0xee, 0xf1,
+        0x22, 0x2a, 0xdb, 0x3d, 0x4f, 0x29, 0xf4, 0x48,
+        0x49, 0x56, 0xb4, 0xc6, 0x89, 0x3b, 0x7e, 0xd7,
+        0x08, 0x7c, 0x2a, 0xb4, 0xcc, 0x6e, 0x4c, 0x12,
+        0x6f, 0xbc, 0xac, 0xfb, 0xb0, 0xe5, 0xf7, 0xc7,
+        0x57, 0xb3, 0x84, 0xa6, 0xb9, 0x2f, 0x59, 0xea,
+        0xaa, 0xf0, 0x9a, 0xe7, 0xff, 0xbc, 0xf8, 0x12,
+        0x09, 0x6e, 0x2b, 0xf8, 0x24, 0xc0, 0x45, 0xb4,
+        0x0c, 0x23, 0x05, 0x7d, 0xf5, 0xc6, 0x8a, 0x19,
+        0x30, 0x8a, 0x40, 0x5a, 0xc6, 0xbd, 0x66, 0xd5,
+        0xdf, 0xc1, 0xbb, 0x8f, 0x26, 0x2d, 0xd1, 0xaf,
+        0x46, 0x47, 0xef, 0x2b, 0x3d, 0x73, 0xd5, 0xcb,
+        0x4a, 0xda, 0xf8, 0x0e, 0xf6, 0xdf, 0x5e, 0x7b
+    };
+
+    rsa = libspdm_rsa_new();
+    if (rsa == NULL) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS rsa new failed \n"));
+        return false;
+    }
+
+    result = libspdm_rsa_set_key(rsa, LIBSPDM_RSA_KEY_N, public_n, sizeof(public_n));
+    if (!result) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS rsa set key failed \n"));
+        libspdm_rsa_free(rsa);
+        return false;
+    }
+
+    result = libspdm_rsa_set_key(rsa, LIBSPDM_RSA_KEY_E, e, sizeof(e));
+    if (!result) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS rsa set key failed \n"));
+        libspdm_rsa_free(rsa);
+        return false;
+    }
+
+    result = libspdm_rsa_set_key(rsa, LIBSPDM_RSA_KEY_D, private_d, sizeof(private_d));
+    if (!result) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS rsa set key failed \n"));
+        libspdm_rsa_free(rsa);
+        return false;
+    }
+
+    result = libspdm_rsa_pkcs1_sign_with_nid(rsa, LIBSPDM_CRYPTO_NID_SHA256,
+                                             input_data, sizeof(input_data),
+                                             signature, &sig_size);
+    if (!result) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS rsa sign failed \n"));
+        libspdm_rsa_free(rsa);
+        return false;
+    }
+
+    /*KAT test*/
+    if (sig_size != sizeof(expected_sig)) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS KAT failed \n"));
+        libspdm_rsa_free(rsa);
+        return false;
+    }
+
+    if (libspdm_const_compare_mem(signature, expected_sig,
+                                  sizeof(expected_sig))) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS KAT failed \n"));
+        libspdm_rsa_free(rsa);
+        return false;
+    }
+
+    result = libspdm_rsa_pkcs1_verify_with_nid(rsa, LIBSPDM_CRYPTO_NID_SHA256,
+                                               input_data, sizeof(input_data),
+                                               signature, sig_size);
+    if (!result) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "RSA_PSS PCT failed \n"));
```

Comment:
remove PCT

Reply:
Thanks. I have changed the `PCT` to `selftest`.